conans/model/editable_cpp_info.py

```diff
@@ -4,54 +4,82 @@
-        data = _parse_func(filepath_or_content, *args, **kwargs)
-        return EditableCppInfo(data)
+        data = _parse_func(filepath_or_content, require_namespace=require_namespace)
+        return EditableCppInfo(data, uses_namespace=require_namespace)
 
     @classmethod
-    def parse_file(cls, filepath, base_path, settings=None, options=None):
-        with open(filepath, 'r') as f:
-            return cls.parse_content(content=six.u(f.read()), base_path=base_path,
-                                     settings=settings, options=options)
+    def parse_file(cls, filepath, require_namespace):
+        with open(filepath) as f:
+            return cls.parse_content(content=six.u(f.read()), require_namespace=require_namespace)
 
     @classmethod
-    def parse_content(cls, content, base_path, settings=None, options=None):
+    def parse_content(cls, content, require_namespace):
         """ Returns a dictionary containing information about paths for a CppInfo object: includes,
         libraries, resources, binaries,... """
-        ret = {k: [] for k in cls.cpp_info_dirs}
-
-        def _work_on_value(value, base_path_, settings_, options_):
-            value = re.sub(r'\\\\+', r'\\', value)
-            value = value.replace('\\', '/')
-            isabs = ntpath.isabs(value) or posixpath.isabs(value)
-            if base_path_ and not isabs:
-                value = os.path.abspath(os.path.join(base_path_, value))
-            value = os.path.normpath(value)
-            value = value.format(settings=settings_, options=options_)
-            return value
 
         parser = configparser.ConfigParser(allow_no_value=True, delimiters=('#', ))
         parser.optionxform = str
         parser.read_string(content)
-        for section in ret.keys():
-            if section in parser:
-                ret[section] = [_work_on_value(value, base_path, settings, options)
-                                for value in parser[section]]
-        return ret
-
-    def apply_to(self, cpp_info):
-        for key, items in self._data.items():
-            setattr(cpp_info, key, items)
-        return cpp_info
+
+        if not require_namespace:
+            ret = {k: [] for k in cls.cpp_info_dirs}
+            for section in ret.keys():
+                if section in parser:
+                    ret[section] = parser[section]
+            return ret
+        else:
+            ret = defaultdict(lambda: {k: [] for k in cls.cpp_info_dirs})
+            for section in parser:
+                if ':' in section:
+                    namespace, key = section.split(':', 1)
+                    if key in cls.cpp_info_dirs:
+                        ret[namespace][key] = parser[section]
+            return ret
+
+    @staticmethod
+    def _work_on_item(value, base_path, settings, options):
+        value = re.sub(r'\\\\+', r'\\', value)
+        value = value.replace('\\', '/')
+        isabs = ntpath.isabs(value) or posixpath.isabs(value)
```

Comment:
This is still extremely weird. I need to understand this path manipulation, probably deserves whiteboard.

Reply:
Tests are passing without the first line, I'm removing it.

About the third line: 
In the general use case there will be one file containing paths for Windows and paths for Linux, and for every non-absolute path we want to calculate the absolute one (prepending the `base_path` varaible). This parsing and normalization is done in Windows and Linux (using the same file) and I think it is cleaner if we identify which paths are already absolute or we will come up with things like this:

```
/this/is/my/base/path/C:/it/was/an/absolute/windows/path
```

I'm even thinking about removing paths that do not correspond to the running OS.